extension/lib/shopgate/CartExtensionPipeline.js

```diff
@@ -49,7 +49,24 @@ class ShopgateCartExtensionPipeline {
       await this._bigCommerceCartRepository.addItems(itemsToAdd)
     }
     if (itemsToUpdate.length) {
-      await this._bigCommerceCartRepository.updateItems(itemsToUpdate, () => {})
+      try {
+        await this._bigCommerceCartRepository.updateItems(itemsToUpdate, () => { })
+      } catch (err) {
+        if (err.code === 422) {
```

Comment:
you will need to move this updateItems call to a function, because you also have `updateProducts` function that makes this call and also needs to be handled. There is also some logic in the updateProducts call that may need to be looked at to see if we are not missing some additional error handling here